File: src/test/scala/gameover/fwk/gdx/GdxTest.scala
package gameover.fwk.gdx

import com.badlogic.gdx.backends.lwjgl.{LwjglApplication, LwjglApplicationConfiguration}
import com.badlogic.gdx.graphics.{GL20, GL30}
import com.badlogic.gdx._
import org.scalamock.scalatest.MockFactory
import org.scalatest._


/**
 * Class to use in test depending on LibGdx. In the test class, initiate LibGdx and pass this
 * application.
 */
object GdxTest{
  def initializeLibGdx(): LwjglApplication = {
    val config = new LwjglApplicationConfiguration()
    config.width = 0
    config.height = 0
    val application: LwjglApplication = new LwjglApplication(new GdxTest, config)
    Gdx.gl = new FakeGL20
    application
  }
}
class GdxTest extends ApplicationAdapter {
}

File: src/test/kotlin/gameover/fwk/libgdx/GdxTest.kt
package gameover.fwk.libgdx

import com.badlogic.gdx.ApplicationAdapter
import com.badlogic.gdx.Gdx
import com.badlogic.gdx.backends.lwjgl.LwjglApplication
import com.badlogic.gdx.backends.lwjgl.LwjglApplicationConfiguration

/**
 * Class to use in test depending on LibGdx. In the test class, initiate LibGdx and pass this
 * application.
 */
class GdxTest : ApplicationAdapter() {

    companion object {
        fun initializeLibGdx(): LwjglApplication {
            val config = LwjglApplicationConfiguration()
            config.width = 0
            config.height = 0
            val application = LwjglApplication(GdxTest(), config)
            Gdx.gl = FakeGL20()
            return application
        }
    }
}